mmseg/apis/train.py

```diff
@@ -70,13 +70,15 @@ def train_segmentor(model,
     # build runner
     optimizer = build_optimizer(model, cfg.optimizer)
 
-    runner = IterBasedRunner(
-        model=model,
-        batch_processor=None,
-        optimizer=optimizer,
-        work_dir=cfg.work_dir,
-        logger=logger,
-        meta=meta)
+    runner = build_runner(
+        cfg.runner,
```

Comment:
This breaks the backward compatibility. It should be better to keep the original api and add a deprecation warning.

Reply:
I was not sure about how to maintain  the backward compatibility so I create a default runner section that reproduces the old behavior.

Reply:
Currently, if the user is using an old config file,  there will be error since `max_iters` is not specified.

Reply:
We may keep the backward compatibility so that user may use old config file.

Reply:
With the block added above:

```python

    if cfg.get('runner') is None:
        cfg.runner = {'type': 'IterBasedRunner', 'max_iters': cfg.total_iters}
        warnings.warn(
            'config is now expected to have a `runner` section, '
            'please set `runner` in your config.', DeprecationWarning)
```
User should be able to use an old config file, right?